receiver/prometheusreceiver/metrics_receiver_test.go

```diff
@@ -195,201 +198,88 @@ buffer_count: 2
 	precv.Flush()
 
 	close(shutdownCh)
-
-	got := cms.AllMetrics()
-
-	// Unfortunately we can't control the time that Prometheus produces from scraping,
-	// hence for equality, we manually have to retrieve the times recorded by Prometheus,
-	// but indexed by each unique MetricDescriptor.Name().
-	retrievedTimestamps := indexTimestampsByMetricDescriptorName(got)
+	gotMDs := cms.AllMetrics()
 
 	// Now compare the received metrics data with what we expect.
-	want1 := []data.MetricsData{
-		{
-			Node: &commonpb.Node{
-				Identifier: &commonpb.ProcessIdentifier{
-					HostName: host,
-				},
-				ServiceInfo: &commonpb.ServiceInfo{
-					Name: "demo",
-				},
-				Attributes: map[string]string{
-					"scheme": "http",
-					"port":   port,
-				},
+	wantNode := &commonpb.Node{
+		Identifier: &commonpb.ProcessIdentifier{
+			HostName:      host,
+			XXX_sizecache: 11,
```

Comment:
How come these `XXX*` fields are getting populated?

Reply:
I'm not sure either. My guess is these fields are added to the gen-go files when they're re-generated by a newer version of proto compiler. Anyway if I don't include these fields, comparison would fail.

Reply:
Perhaps you'll need to examine how to fix these comparison options in https://github.com/census-instrumentation/opencensus-service/pull/539/files#diff-b62f24d39c37eacaa5d4545576874137R319 perhaps with `proto.Equal` otherwise anyone using this new comparison helper is going to have to add those `XXX` fields

Reply:
Just tried `proto.Equal`, unfortunately by using it we need to know the exact `timestamp` of the given metrics, which is also tedious. I'll try another way around.